### ISSUE TITLE
unexpected behavior in cv::divide

### ISSUE THREAD

**Issue by pasbi** [opencv/opencv#14995]
##### System information (version)
- OpenCV 4.1.0
- Arch Linux (x64)

##### Detailed description

It's about the `divide(double scale, InputArray src2, OutputArray dst, int dtype)` overload of the `divide` function.
Only the first channel of `dst` is computed as expected; the other channels are always zero.

##### Steps to reproduce
- Python:
`import cv2; import numpy as np; cv2.divide(1, np.ones((1, 1, 2)))`
The output is `array([[[1., 0.]]])` but I expected `array([[[1., 1.]]])`.

##### Solution
- change the type of the first argument's (`scale`) from `double` to `cv::Scalar`
- apply the `scale`-argument to each channel. That's what numpy already does: `1.0/np.ones((1, 1, 2)) == array([[[1., 1.]]])`
- Just in case that this is indeed the desired behavior: mention it in the documentation explicitly. I believe that currently, the documentation says that `cv::divide` works like `numpy` ("[Different channels of multi-channel arrays are processed independently.](https://docs.opencv.org/4.1.0/d2/de8/group__core__array.html#ga6db555d30115642fedae0cda05604874)"). However, I find it somewhat ambiguous.

**Comment by alalek** [comment]
`double scale` is `Scalar(scale, 0, 0, 0)`. It is not `Scalar(scale, scale, scale, scale)`.

This can't be changes with breaking of users code (so it would not be changed).

```.diff
-cv2.divide((1), np.ones((1, 1, 2))
+cv2.divide((1,1,1,1), np.ones((1, 1, 2))
```

---

Usage questions should go to Users OpenCV Q/A forum: http://answers.opencv.org

**Comment by pasbi** [comment]
This is not a usage question. Now I know how to use it, or at least how to work around the issue. My actual problem was solved before I raised this issue.

I just want to improve the code, or, if you think that's not possible because of legacy, at least the documentation.
The current behavior is very unexpected and the documentation is at best ambiguous. But I find it misleading.

I don't understand your argument. Maybe I was not clear enough:

1.  The best solution IMHO is to change the behavior of the `cv::divide(double, cv::InputArray, int)` overload to be consistent with what numpy does.
2. if (1.) is not possible (which I'd understand totally), add another overload `cv::divide(cv::Scalar, cv::InputArray, int)` and add a sentence to the documentation that `cv::divide(double, cv::InputArray, int)` does not do what one probably expects for multi-channel images.

There are three workarounds in C++:
1. iterate manually (two nested loops)
2. split the multi-channel image, divide each channel individually, then merge the channels again
3. create a image of ones of same shape and use the other overload.

Each of these solutions is bad because of either performance or readability reasons.
